File: May 2022/May_06.cpp
class Solution {
public:
    string removeDuplicates(string s, int k) {
        int n = s.length();
        stack<char> st1;
        stack<int> st2;
        for(int i = 0; i < n; i++){
            if(!st1.empty() && st1.top() == s[i])
                st2.push(st2.top() + 1);
            else
                st2.push(1);
            st1.push(s[i]);
            if(st2.top() == k){
                for(int i = 0; i < k; i++)
                    st1.pop(), st2.pop();
            }
        }
        string str = "";
        while(!st1.empty())
            str += st1.top(), st1.pop();
        reverse(str.begin(), str.end());
        return str;
    }
};

File: May 2022/May_06.java
class Solution {
    public String removeDuplicates(String s, int k) {
        int n = s.length();
        Stack<Character> st1 = new Stack<>();
        Stack<Integer> st2 = new Stack<>();
        for(int i = 0; i < n; i++){
            if(!st1.isEmpty() && st1.peek() == s.charAt(i))
                st2.push(st2.peek() + 1);
            else
                st2.push(1);
            st1.push(s.charAt(i));
            if(st2.peek() == k){
                for(int j = 0; j < k; j++){
                    st1.pop();
                    st2.pop();
                }
            }
        }
        StringBuilder sb = new StringBuilder();
        while(!st1.isEmpty())
            sb.append(st1.pop());
        return sb.reverse().toString();
    }
}
